### PR TITLE
fix for WhenAny WhenAnyValue ToProperty

### LINKED ISSUE #2812: [BUG] Version 14.1.1 WhenAny.. now returns observable to a nullable regardless of parameter type
**Describe the bug**
As a result of upgrading to version 14.1.1 with nullable references enabled in a .net 5.0 wpf project, regardless of the input type,
whenAny.. now returns a nullable reference type.  This means regardless of the input parameters, we have to use WhereNotNull() everywhere or bangs (I reserve them for last resort).  I believe this change came through in issue 2760

**Expected behavior**
The return type should reflect the input value types.

### PR DESCRIPTION
**Pull request by ChrisPulman**
**What kind of change does this PR introduce?**

update  WhenAny WhenAnyValue to use nullable for selector
fix #2812 ToProperty to work with nullable OAPH and non nullable OAPH

**What is the current behaviour?**

causes compiler issue or runtime exception

**What is the new behaviour?**

WhenAnyValue and ToProperty can be used in conjunction with each other without compiler warnings or runtime exceptions.

**What might this PR break?**

nullablilty of OAPH needs to be checked.

**Please check if the PR fulfils these requirements**
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Other information**:

### CODE REVIEW

File: src/ReactiveUI.Splat.Tests/ReactiveUI.Splat.Tests.csproj [hunk 1]
```diff
@@ -9,9 +9,9 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Splat.Autofac" Version="12.*" />
+    <PackageReference Include="Splat.Autofac" Version="11.*" />
```
Comment (glennawatson) [inline src/ReactiveUI.Splat.Tests/ReactiveUI.Splat.Tests.csproj:12]: dont we want 12.*?

File: src/ReactiveUI/ReactiveUI.csproj [hunk 2]
```diff
@@ -22,7 +22,7 @@
   <ItemGroup>
     <Compile Remove="Platforms\**\*.cs" />
     <None Include="Platforms\**\*.cs" />
-    <PackageReference Include="Splat" Version="12.*" />
+    <PackageReference Include="Splat" Version="11.*" />
```
Comment (glennawatson) [inline src/ReactiveUI/ReactiveUI.csproj:25]: again v12?

File: src/ReactiveUI.Splat.Tests/ReactiveUI.Splat.Tests.csproj (hunk 1 above)
Comment (ChrisPulman) [reply src/ReactiveUI.Splat.Tests/ReactiveUI.Splat.Tests.csproj:12]: Absolutely the final version will have V12

File: src/ReactiveUI/ReactiveUI.csproj (hunk 2 above)
Comment (ChrisPulman) [reply src/ReactiveUI/ReactiveUI.csproj:25]: Absolutely the final version will have V12

**Review by glennawatson — APPROVED** [review]

**Pull request merged**